server/routerlicious/packages/routerlicious-base/src/nexus/api/api.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { validateRequestParams } from "@fluidframework/server-services-shared";
+import {
+	throttle,
+	IThrottleMiddlewareOptions,
+	verifyStorageToken,
+} from "@fluidframework/server-services-utils";
+import * as core from "@fluidframework/server-services-core";
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+import {
+	IBroadcastSignalEventPayload,
+	ICollaborationSessionEvents,
+	IRoom,
+	IRuntimeSignalEnvelope,
+} from "@fluidframework/server-lambdas";
+import { Router } from "express";
+import winston from "winston";
+import { Provider } from "nconf";
+import { Constants } from "../../utils";
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+
+export function create(
+	config: Provider,
+	tenantManager: core.ITenantManager,
+	tenantThrottlers: Map<string, core.IThrottler>,
+	storage: core.IDocumentStorage,
+	collaborationSessionEventEmitter?: TypedEventEmitter<ICollaborationSessionEvents>,
+): Router {
+	const router: Router = Router();
+
+	const tenantThrottleOptions: Partial<IThrottleMiddlewareOptions> = {
+		throttleIdPrefix: (req) => req.params.tenantId,
+		throttleIdSuffix: Constants.nexusRestThrottleIdSuffix,
+	};
+	const generalTenantThrottler = tenantThrottlers?.get(Constants.generalRestCallThrottleIdPrefix);
+
+	router.post(
+		"/:tenantId/:id/broadcast-signal",
+		validateRequestParams("tenantId", "id"),
+		throttle(generalTenantThrottler, winston, tenantThrottleOptions),
+		verifyStorageToken(tenantManager, config),
+		// eslint-disable-next-line @typescript-eslint/no-misused-promises
+		async (request, response) => {
+			const tenantId = request.params.tenantId;
+			const documentId = request.params.id;
+			const signalContent = request?.body?.signalContent;
+			if (!isValidSignalEnvelope(signalContent)) {
+				response
+					.status(400)
```

Comment:
We have a lot of extra functionality baked into the `NetworkError` class and `handleResponse` method. Instead of manually calling `response.status().send()`, it would be preferable to extract all this logic into a helper method, like `async function handleBroadcastSignal(...)`, then pass the result into `handlResponse()`. Then, within `handleBroadcastSignal`, replace all these `res.status.send; return`s with `throw NetworkError(code, message)`